metric_learn/base_metric.py

```diff
@@ -59,7 +59,7 @@ def fit(self, X, y):
 
 class WeaklySupervisedMetricLearner(BaseMetricLearner):
 
-  def fit(self, X, constraints):
+  def fit(self, constrained_dataset, y):
```

Comment:
maybe think of a more concise naming convention instead of `constrained_dataset` as it is gonna be used all over the place. `X_constrained` perhaps?

Reply:
I agree, ``X_constrained`` seems good